python-sdk/tests/test_transaction.py

```diff
@@ -84,3 +85,19 @@ def test_delete_transaction(client: TestClient, default_project: bool):
 
         with pytest.raises(grpc.RpcError):
             client.transaction.delete_transaction(transaction=txn)
+
+    @staticmethod
+    def test_file_operations_within_transaction(client: TestClient, default_project: bool):
+        """Ensure that file operations """
+        repo = client.new_repo(default_project)
+        branch = pfs.Branch(repo=repo, name="master")
+        test_file = pfs.File.from_uri(f"{repo.project.name}/{repo.name}@master:/file.dat")
+
+        with client.transaction.transaction() as _txn:
+            repo_txn = client.new_repo(default_project)
+            with client.pfs.commit(branch=branch) as commit:
+                with pytest.raises(InvalidTransactionOperation):
+                    commit.put_file_from_bytes(path=test_file.path, data=b"hello world")
```

Comment:
up to you whether or not you want to implement this, but coverage for the other write operations seems lacking. it's going to be tedious to get them all but maybe adding copy/delete as well as a put would be a good idea